sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -59,31 +65,120 @@ public HttpPipeline getHttpPipeline() {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin)
+            .flatMap(
+                response -> Mono.just(mapper.convertValue(response.getValue(), klazz)));
+    }
+
+    // Input is String and output is Response<String>.
+    // TODO: Autorest team -> the etag returned by the service is present under both Response.getHeaders() and ResponseBase.deserializedHeaders().
+    // TODO: (cont.) Since etag is a well known http header, it should be available via Response.getHeaders(), which it is.
+    // TODO: (cont.) So there shouldn't be a need to define DigitalTwinsAddHeaders explicitly again, and map it to ResponseBase.deserializedHeaders.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<String>> createDigitalTwinWithResponseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        return Mono.just(new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is String and output is Response<String> -> ResponseBase<DigitalTwinsAddHeaders, String>.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ResponseBase<DigitalTwinsAddHeaders, String>> createDigitalTwinWithResponseResponseBaseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        return Mono.just(new ResponseBase<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, response.getDeserializedHeaders()));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is String and output is Response<String> -> DigitalTwinsAddResponse (json string).
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsAddResponse> createDigitalTwinWithResponseDigitalTwinAddResponseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
```

Comment:
Option 3:
Accept request as string, deserialize it to object and send to PL.
PL returns response as an object, serialize it to string and return to user.

Response type returned is an implementation of `ResponseBase<H, T>` - it contains the etag info in explicitly defined `deserializedHeaders`.

This response type looks clunky, and adds no extra value.

Reply:
The only difference in retrieving the etags:

`Response<T>` - response.getHeaders().get("etag")
`Response<H, T>` - response.getDeserializedHeaders().getETag()